cpp/src/arrow/util/cpu_info.h

```diff
@@ -59,9 +62,11 @@ class ARROW_EXPORT CpuInfo {
   /// Returns whether of not the cpu supports this flag
   bool IsSupported(int64_t flag) const { return (hardware_flags_ & flag) != 0; }
 
+#ifdef ARROW_HAVE_SSE4_2
```

Comment:
Instead of doing this, you should just return false from `CanUseSSE4_2` if `ARROW_HAVE_SSE4_2` is disabled.